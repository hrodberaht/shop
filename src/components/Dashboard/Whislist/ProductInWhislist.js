import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class ProductInWhislist extends Component {
  handleClick = () => {
    const {
      remove,
      product: { productId },
    } = this.props;
    remove(productId);
  };

  render() {
    const {
      product: {
        name, price, pcsOrder, totalPrice,
      },
    } = this.props;
    return (
      <tr>
        <td>{name}</td>
        <td>{price}</td>
        <td>{pcsOrder}</td>
        <td>
          <span>$</span>
          {totalPrice}
        </td>
        <td>
          <button type="button" className="btn btn-danger" onClick={this.handleClick}>
            X
          </button>
        </td>
      </tr>
    );
  }
}

ProductInWhislist.propTypes = {
  product: PropTypes.shape({
    name: PropTypes.string,
    type: PropTypes.string,
    price: PropTypes.number,
  }).isRequired,
  remove: PropTypes.func.isRequired,
};
