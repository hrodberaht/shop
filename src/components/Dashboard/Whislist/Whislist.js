import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { fetchWhislist, removeProductWhisList } from '../../../store/whislist/actionCerator';
import { getAuthToken, getAuthUserId } from '../../../store/auth/selectors';
import { getWhislistLoaded, getWhislistProducts } from '../../../store/whislist/selectors';
import ProductInWhislist from './ProductInWhislist';

export class Whislist extends Component {
  componentDidMount() {
    const { getWhislist, userId, token } = this.props;
    getWhislist(userId, token);
  }

  remove = (product) => {
    const { removeProd, userId, token } = this.props;
    removeProd(product, userId, token);
  };

  render() {
    const { loaded, products } = this.props;
    if (!loaded) return <h4>Loading</h4>;
    return (
      <div>
        <table>
          <tbody>
            {products.map(prod => (
              <ProductInWhislist product={prod} remove={this.remove} key={prod.productId} />
            ))}
          </tbody>
        </table>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  token: getAuthToken(state),
  userId: getAuthUserId(state),
  loaded: getWhislistLoaded(state),
  products: getWhislistProducts(state),
});

export default connect(
  mapStateToProps,
  {
    getWhislist: fetchWhislist,
    removeProd: removeProductWhisList,
  },
)(Whislist);

Whislist.propTypes = {
  getWhislist: PropTypes.func,
  userId: PropTypes.string,
  token: PropTypes.string.isRequired,
  loaded: PropTypes.bool.isRequired,
  products: PropTypes.arrayOf(PropTypes.objectOf),
  removeProd: PropTypes.func.isRequired,
};

Whislist.defaultProps = {
  products: null,
  getWhislist: null,
  userId: null,
};
