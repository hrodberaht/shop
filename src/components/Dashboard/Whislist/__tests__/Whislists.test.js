import { Whislist } from '../Whislist';

const defaultProps = {
  products: [
    {
      id: 'product-1',
      type: 'Printer',
      name: 'ZX3',
      price: '400',
      inStock: '30',
    },
    {
      id: 'product-2',
      type: 'Printer',
      name: 'ZX5',
      price: '600',
      inStock: '5',
    },
  ],
  getWishlist: jest.fn(),
  token: '12345',
  loaded: false,
};

const setup = buildSetup(Whislist, defaultProps);

describe('<Whislist />', () => {
  it('should render without crash', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  //   it('should render products list if loaded true', () => {
  //     const { wrapper } = setup();
  //     wrapper.setProps({ loaded: true });
  //     expect(wrapper).toMatchSnapshot();
  //   });

  //   it('should call handleFetch with token', () => {
  //     const { wrapper } = setup();
  //     expect(wrapper.instance().props.handleFetch).toHaveBeenCalledWith('12345');
  //   });
});
