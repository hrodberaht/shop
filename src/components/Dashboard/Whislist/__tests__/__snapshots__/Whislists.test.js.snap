// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Wishlist /> should render products list if loaded true 1`] = `
<div>
  <h2>
    Wish list:
  </h2>
  <table>
    <tbody>
      <ProductInWhislist
        key="product-1"
        product={
          Object {
            "inStock": "30",
            "name": "ZX3",
            "price": 400,
            "productId": "product-1",
            "type": "Printer",
          }
        }
        remove={[Function]}
      />
      <ProductInWhislist
        key="product-2"
        product={
          Object {
            "inStock": "5",
            "name": "ZX5",
            "price": 600,
            "productId": "product-2",
            "type": "Printer",
          }
        }
        remove={[Function]}
      />
    </tbody>
  </table>
</div>
`;

exports[`<Wishlist /> should render without crash 1`] = `
<h4>
  Loading
</h4>
`;
